roles/pulp_workers/tasks/main.yml

```diff
@@ -9,7 +9,36 @@
   notify: Reload systemd and restart pulp workers
   become: true
 
-- name: Set states of pulp workers
+- name: Get services
+  service_facts:
+
+- name: Get already existing workers
+  set_fact:
+    worker_services: '{{ worker_services|default([]) + [ item | regex_search("(?<=pulpcore-worker@)\d+(?=\.service$)") ] }}'
```

Comment:
Let's change the fact name to __pulp_workers_worker_svcs to avoid any possible conflicts with later roles the users may add. (e.g., a role for another application also on the pulp server.)